src/Lookup/InMemoryEntityLookup.php

```diff
@@ -18,6 +21,7 @@
 class InMemoryEntityLookup implements EntityLookup {
 
 	private $entities = array();
+	private $exceptions = array();
```

Comment:
If you start by looking at private implementation details rather than the public interface, then yeah. No need to read the whole file - just look at the value writes, which is one location, which has the type on the line above it.

By all means feel free to add the type hints though. Somewhat cluttery to me, though not an issue.